src/main/java/org/apache/sling/event/impl/jobs/JobHandler.java

```diff
@@ -78,91 +78,77 @@ public boolean startProcessing(final Queue queue) {
+                    final Map<String, Object> props = new HashMap<>(mvm);
+                    props.put(JobImpl.PROPERTY_FINISHED_STATE, state.name());
+                    if (isSuccess) {
+                        // we set the finish date to start date + duration
+                        final Date finishDate = new Date();
+                        finishDate.setTime(job.getProcessingStarted().getTime().getTime() + duration);
+                        final Calendar finishCal = Calendar.getInstance();
+                        finishCal.setTime(finishDate);
+                        props.put(JobImpl.PROPERTY_FINISHED_DATE, finishCal);
+                    } else {
+                        // current time is good enough
+                        props.put(JobImpl.PROPERTY_FINISHED_DATE, Calendar.getInstance());
+                    }
+                    if (job.getProperty(Job.PROPERTY_RESULT_MESSAGE) != null) {
+                        props.put(Job.PROPERTY_RESULT_MESSAGE, job.getProperty(Job.PROPERTY_RESULT_MESSAGE));
                     }
-                    resolver.delete(jobResource);
-                    resolver.commit();
+                    ResourceHelper.getOrCreateResource(rr, newPath, props);
+                }
+                rr.delete(jobResource);
+                rr.commit();
 
-                    if ( keepJobInHistory && configuration.getMainLogger().isDebugEnabled() ) {
-                        if ( isSuccess ) {
-                            configuration.getMainLogger().debug("Kept successful job {} at {}", Utility.toString(job), newPath);
-                        } else {
-                            configuration.getMainLogger().debug("Moved cancelled job {} to {}", Utility.toString(job), newPath);
-                        }
+                if (keepJobInHistory && configuration.getMainLogger().isDebugEnabled()) {
+                    if (isSuccess) {
+                        configuration.getMainLogger().debug("Kept successful job {} at {}", Utility.toString(job),
+                                newPath);
+                    } else {
+                        configuration.getMainLogger().debug("Moved cancelled job {} to {}", Utility.toString(job),
+                                newPath);
                     }
-                } catch ( final PersistenceException pe ) {
-                    this.configuration.getMainLogger().warn("Unable to finish job " + job.getId(), pe);
-                } catch (final InstantiationException ie) {
-                    // something happened with the resource in the meantime
-                    this.configuration.getMainLogger().debug("Unable to instantiate job", ie);
                 }
+            } catch (final PersistenceException pe) {
+                this.configuration.getMainLogger().warn("Unable to finish job " + job.getId(), pe);
             }
-        } finally {
-            resolver.close();
-        }
+            return false;
```

Comment:
IIUC this `return false` is a no-op here, as the return value is not really evaluated by the caller. 

Yet I was wondering if there's a particular reason to return `false` rather than `true`? (or should we add a comment that states that "this return value is not used anywhere"?)

Reply:
```persistJobProperties``` and ```reschedule``` return a boolean, so it has to be a Function, if I want to use the same code in all cases.
I will add statements to the Lambdas to indicate that this value is actually ignored.